ospfd/ospf_opaque.c

```diff
@@ -427,11 +427,10 @@ void ospf_delete_opaque_functab(uint8_t lsa_type, uint8_t opaque_type)
 				if (functab->oipt != NULL)
 					free_opaque_info_per_type(functab->oipt,
 								  true);
-				/* Dequeue listnode entry from the function table
-				 * list coreesponding to the opaque LSA type.
-				 * Note that the list deletion callback frees
-				 * the functab entry memory.
-				 */
+				else
+					ospf_opaque_del_functab(functab);
```

Comment:
I see there is a problem as only list_delete_all_node(), i.e., deleting the entire list at once via list_delete(),  invokes the list->del function. What a crock.. However, this fix isn't right as you don't want to free the functab memory before removing it from the list. Let me look at this logic a bit more as now I'm very familiar with the data structures.

Reply:
Please try this diff - it works for me and there is no memory leak or double free. 
[ospf_opaque.txt](https://github.com/FRRouting/frr/files/13587838/ospf_opaque.txt)

Reply:
I have tested the diff you provided and it works like charm. Thanks a bunch for the fix!

Reply:
There appears to be a heap-use-after-free error when `test_ospf_gr_topo1.py` is run due to oipt still holding reference to a deleted `functab`. Any insights on how we can address this issue?